WcaOnRails/app/models/competition.rb

```diff
@@ -441,15 +441,15 @@ def results_uploaded?
   end
 
   def events_with_podium_results
-    results.where(roundId: Round.final_rounds.map(&:id)).where("pos >= 1 AND pos <= 3").group_by(&:event).sort_by { |event, results| event.rank }
+    light_results_from_relation(results.where(roundId: Round.final_rounds.map(&:id)).where("pos >= 1 AND pos <= 3")).group_by(&:event).sort_by { |event, results| event.rank }
   end
 
   def winning_results
-    results.where(roundId: Round.final_rounds.map(&:id)).where("pos = 1").sort_by { |r| r.event.rank }
+    light_results_from_relation(results.where(roundId: Round.final_rounds.map(&:id)).where("pos = 1")).sort_by { |r| r.event.rank }
   end
 
   def persons_with_results
```

Comment:
nit: I believe that this function should be renamed. How about the more explicit `person_names_with_results`?